Rollershutter potential nil error on operations

@pacive  - 
  I was looking at the rollershutter code and it is different than the others in that I don't think it handles the case where the rollershutter state is UNDEF or NULL and I don't see a test for either of those conditions.

  state can return nil (and does for UNDEF or NULL) so all of your checks should be done with the safe navigation operator for up?/down? and position.

> For your specific use case, I am curious if this works?
> 
> ```
> gBatteries.reject{ |item| item.state&. > LOW_BATTERY_THRESHOLD}
> ```

Resulted in an NPE
```
21:04:28.971 [ERROR] [ript.internal.ScriptEngineManagerImpl] - Error during evaluation of script 'file:/openhab/conf/automation/jsr223/ruby/personal/battery.rb': org.jruby.embed.EvalFailedException: java.lang.NullPointerException
```



> gBatteries.grep(UNDEF) + gBatteries.grep_v(UNDEF).select {|item| item < LOW_BATTERY_THRESHOLD).sort

```
21:06:03.908 [ERROR] [ript.internal.ScriptEngineManagerImpl] - Error during evaluation of script 'file:/openhab/conf/automation/jsr223/ruby/personal/battery.rb': org.jruby.embed.EvalFailedException: (NoMethodError) undefined method `to_big_decimal' for nil:NilClass
Did you mean?  BigDecimal
```



Sorry if this is off topic, shall I open a separate issue?

> gBatteries.grep(UnDefType) + gBatteries.grep_v(UnDefType).select { |item| item < LOW_BATTERY_THRESHOLD }.sort

```
01:35:08.141 [ERROR] [ript.internal.ScriptEngineManagerImpl] - Error during evaluation of script 'file:/openhab/conf/automation/jsr223/ruby/personal/test.rb': org.jruby.embed.EvalFailedException: (NoMethodError) undefined method `to_big_decimal' for nil:NilClass
Did you mean?  BigDecimal
```

--- a/features/rollershutter_item.feature
+++ b/features/rollershutter_item.feature
@@ -113,6 +113,8 @@ Feature: rollershutter_item
       when DOWN then logger.info('RollerOne is DOWN')
       when (51...100) then logger.info('RollerOne is in range 51...100')
       when (1..50) then logger.info('RollerOne is in range 1..50')
+      else
+        logger.info('RollerOne is NULL')
       end
       """
     When I deploy the rules file
@@ -122,3 +124,4 @@ Feature: rollershutter_item
       | 0       | UP             |
       | 100     | DOWN           |
       | 25      | in range 1..50 |
+      | NULL    | NULL           |

--- a/lib/openhab/dsl/items/rollershutter_item.rb
+++ b/lib/openhab/dsl/items/rollershutter_item.rb
@@ -43,7 +43,7 @@ module OpenHAB
         # @return [Java::OrgOpenhabCoreLibraryTypes::PercentType] the position of the rollershutter
         #
         def position
-          state.as(PercentType)
+          state&.as(PercentType)
         end
 
         #
@@ -54,6 +54,8 @@ module OpenHAB
         # @return [Integer] -1, 0 or 1 depending on the result of the comparison
         #
         def <=>(other)
+          return nil unless state?
+
           case other
           when PercentType, Java::OrgOpenhabCoreLibraryTypes::DecimalType then position.compare_to(other)
           when Numeric then position.int_value <=> other
@@ -73,8 +75,8 @@ module OpenHAB
           raise ArgumentError, "Cannot coerce to #{other.class}" unless other.is_a? Numeric
 
           case other
-          when Integer then [other, position.int_value]
-          when Float then [other, position.float_value]
+          when Integer then [other, position&.int_value]
+          when Float then [other, position&.float_value]
           end
         end
 
@@ -84,7 +86,7 @@ module OpenHAB
         %i[+ - * / %].each do |operator|
           define_method(operator) do |other|
             right, left = coerce(other)
-            left.send(operator, right)
+            left&.send(operator, right)
           end
         end
       end

